File: linked_lists/get_middle.java
public class ListNode {
    int val;
    ListNode next;
    ListNode (int val) {
        this.val = val;
    }
}

int getMiddle(ListNode head)
{
    ListNode slow = head;
    ListNode fast = head;

    while (fast != null && fast.next != null)
    {
        slow = slow.next;
        fast = fast.next.next;
    }

    return slow.val;
}

File: linked_lists/get_middle.js
class ListNode {
    constructor(val) {
        this.val = val;
        this.next = null;
    }
}

const getMiddle = head => {
    let slow = head;
    let fast = head;

    while (fast && fast.next)
    {
        slow = slow.next;
        fast = fast.next.next;
    }

    return slow.val;
}